src/state/messages/selectors.js

```diff
@@ -1 +1,34 @@
 import { createSelector } from 'reselect';
+import { last } from 'lodash';
+
+export const messagesSelector = state => state.messages.messages;
+export const conversationDetailsSelector = state => state.messages;
+
+export const selectLastMessage = createSelector(
+  messagesSelector,
+  message => last(message)
+);
+
+export const selectLastMessageTimestamp = createSelector(
+  selectLastMessage,
+  (selectLastMessage) => selectLastMessage.timestamp
+);
+
+export const selectActiveConversationId = createSelector(
+  conversationDetailsSelector,
+  messages => messages.id
+);
+
+export const selectConversationist = state => state.messages.conversetionist;
+
+const findUsername = (userId, users) => {
+  const messageSender = users.find(user => user.id === userId);
+  return messageSender.username;
+};
+
+export const addUsernamesToMessage = (state, messages) => {
```

Comment:
Maybe something like:
```
const selectDetailedMessages = createSelector(
  [ selectUsers, selectMessages ],
  (users, messages) => messages.map(message => ({
    ...message,
    username: users.find(user => user.id === message.userId)
  }));
);
```

What do you think?

Reply:
Yeah, better, but unfortunately not applicable in this case or I do sth wrong.. I have left it for now, but i will be think about it